controllers/progressiverollout_controller_test.go

```diff
@@ -58,11 +64,50 @@ var _ = Describe("ProgressiveRollout Controller", func() {
 				}},
 			},
 		}
-		Expect(k8sClient.Create(ctx, &pr)).To(Succeed())
+		Expect(k8sClient.Create(ctx, singleStagePR)).To(Succeed())
+	})
 
-		expected := pr.NewStatusCondition(deploymentskyscannernetv1alpha1.CompletedCondition, metav1.ConditionTrue, deploymentskyscannernetv1alpha1.StagesCompleteReason, "All stages completed")
-		ExpectCondition(&pr, expected.Type).Should(HaveStatus(expected.Status, expected.Reason, expected.Message))
+	AfterEach(func() {
+		err := k8sClient.Delete(context.Background(), ns)
+		Expect(err).NotTo(HaveOccurred(), "failed to delete test namespace")
+	})
+
+	Describe("requestsForApplicationChange function", func() {
+		It("should forward events for owned applications", func() {
+			By("creating an owned application")
+			ownedApp := &argov1alpha1.Application{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app",
+					Namespace: namespace,
+					OwnerReferences: []metav1.OwnerReference{{
+						APIVersion: argoApiGroup,
+						Kind:       appSetKind,
+						Name:       "single-stage-appset",
+						UID:        uuid.NewUUID(),
+					}},
+				},
+				Spec: argov1alpha1.ApplicationSpec{},
+			}
+			Expect(k8sClient.Create(ctx, ownedApp)).To(Succeed())
+
+			requests := reconciler.requestsForApplicationChange(ownedApp)
+			Expect(len(requests)).To(Equal(1))
```

Comment:
This is a flaky test because sometimes `len(requests)` is 2. I need to investigate why and I suspect because I'm doing something wrong with Ginkgo.

Reply:
Solved the flakyness by assigning the ownership only in the required test.